### ISSUE TITLE
SwitchIB support and issues

### ISSUE THREAD

**Issue by keenandr** [stanford-rc/ibswinfo#5]
Hi,

The unmanaged EDR Switch-IB based switches (SB7790) appear to mostly work. The PSID is reported incorrectly; it appears to be transposed?

```
# flint -d /dev/mst/SW_MT52000_SwitchIB_Mellanox_Technologies_lid-0x0005 q | grep PSID
PSID:                  HP_1880110032
```
versus

```
# ./ibswinfo.sh -d lid-5 | grep PSID
PSID               | 1_PH108830012
```

It also only reports 35 instead of 36 ports.

**Comment by kcgthb** [comment]
Thanks for reporting this!

I unfortunately don't have any SB7790 model to test directly, but the first thing that comes to mind is that `HP_1880110032` is not a Mellanox PSID, but rather an HPE one. It's possible that the HPE firmware for those switches behaves differently, in ways that could result in the register addresses being shifted.

Could you send me the output of the following commands?
```
# mlxreg -d lid-5 --reg_name MGIR --get
# mlxreg -d lid-5 --reg_name MGPIR --get
# smpquery NI 5
```

**Comment by kcgthb** [comment]
Thank you for the details you provided by email!
Both issues should be fixed in 0.3
